### PR TITLE
Manual test that we get the right values from the FFI on upload

### PR DESCRIPTION
**Pull request by badboy**

### CODE REVIEW

**Comment by badboy** [comment]
draft so I don't forget to: 
* add a changelog entry
* probably adjust a comment or two

File: glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt [hunk 1]
```diff
@@ -27,7 +27,7 @@ enum class UploadTaskTag {
 @Structure.FieldOrder("tag", "documentId", "path", "body", "headers")
 internal class UploadBody(
     // NOTE: We need to provide defaults here, so that JNA can create this object.
-    @JvmField val tag: Byte = UploadTaskTag.Done.ordinal.toByte(),
+    @JvmField val tag: Byte = UploadTaskTag.Upload.ordinal.toByte(),
```
Comment (brizental) [inline glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt:30]: Why was it necessary to change the default here?

File: glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt [hunk 2]
```diff
@@ -43,11 +43,18 @@ internal class UploadBody(
     }
 }
 
+@Structure.FieldOrder("tag", "time")
+internal class WaitBody(
+    @JvmField var tag: Byte = UploadTaskTag.Wait.ordinal.toByte(),
+    @JvmField var time: Long = 60_000
+) : Structure() {
+}
+
```
Comment (brizental) [inline glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt:52]: Sheesh, what an oversight. Good thing you caught this...

**Review by brizental — APPROVED** [review]
Looks good to me.

File: glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt (hunk 1 above)
Comment (badboy) [reply glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt:30]: It wasn't. I just noticed it when I wrote the `WaitBody` and IMO it felt wrong to have a tag of `Done` on an `UploadBody`.
We need default values, but won't actually use a default-constructed `UploadBody` further, so this change is not really noticable.

File: glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt (hunk 2 above)
Comment (badboy) [reply glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt:52]: I didn't notice in review either. That's why we do the testing.

**Pull request merged**